crates/dataflow-jit/src/codegen/json/deserialize.rs

```diff
@@ -51,8 +52,8 @@ pub unsafe fn call_deserialize_fn(
     } else {
         #[inline(never)]
         #[cold]
-        fn deserialize_error(error: String) -> Box<dyn std::error::Error> {
-            format!("an error occurred while parsing the key {error}").into()
+        fn deserialize_error(error: String) -> AnyError {
+            anyhow!(format!("an error occurred while parsing the key {error}"))
```

Comment:
```suggestion
            anyhow!("an error occurred while parsing the key {error}")
```

Reply:
In fact, I should just be outputting `error`, it's up to the caller to build an error message with it.